File: backend/stock-serv/src/main/java/org/willingfish/stock/serv/pojo/DealingRecord.java
package org.willingfish.stock.serv.pojo;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Builder;
import lombok.Data;
import org.willingfish.stock.common.pojo.Money;
import org.willingfish.stock.common.pojo.Stock;

import java.time.LocalDate;

@Data
@Builder
public class DealingRecord {

    Stock stock;
    Money incomePrice;
    Money outcomePrice;
    @JsonFormat(pattern = "yyyy/MM/dd")
    LocalDate incomeTime;
    @JsonFormat(pattern = "yyyy/MM/dd")
    LocalDate outcomeTime;
}

File: backend/stock-serv/src/main/kotlin/org/willingfish/stock/serv/pojo/DealingRecord.kt
package org.willingfish.stock.serv.pojo

import com.fasterxml.jackson.annotation.JsonFormat
import org.willingfish.stock.common.pojo.Money
import org.willingfish.stock.common.pojo.Stock
import java.time.LocalDate

data class DealingRecord(
        val id: Long?,
        val stock: Stock?=null,
        val incomePrice:Money?=null,
        val outcomePrice:Money?=null,
        @JsonFormat(pattern = "yyyy/MM/dd") val incomeTime:LocalDate?=null,
        @JsonFormat(pattern = "yyyy/MM/dd") val outcomeTime:LocalDate?=null
        )
